### DISCUSSION TITLE
Various formulae requiring macFUSE fail installation with dependency error although macFUSE is installed

### DISCUSSION THREAD

**Discussion by porg** [Homebrew/discussions#1760] [Everyday usage] [answered]
## Environment
- macOS 11.4 Big Sur
- brew 3.2.0
- macFUSE 4.1.2

## Problem
- Yesterday I filed:
   libfuse/sshfs/issues/255
   **Note:** Has a lot of details and also working workaround!
- And also other users who attempt to install plugins for macFUSE fail similarly, i.e.
  telepresenceio/telepresence/issues/1654
- And today I wanted to `$ brew install curlftpfs` and again failed with an error message of the same fashion:
  `curlftpfs has been disabled because it requires FUSE!`

## Common problem observed
In [brew](https://brew.sh) all packages with a macFUSE dependency fail installing under the current OS and version:
- https://github.com/telepresenceio/telepresence
-  https://github.com/libfuse/sshfs
- https://github.com/JackSlateur/curlftpfs

## Followup
- I have no idea what the common cause is, where the issue is exactly.
  - In brew?
    - Maybe in macOS Big Sur's raised security handling regarding Kernel Extensions?
  - In the macFUSE package definitions on brew?
    - Note: Earlier it was called osxfuse, maybe some old/new name reference issues?
  - In the brew package definitions of all the macFUSE plugins?

**Comment by carlocab** [comment, marked as answer]
[macFUSE is no longer open source](https://github.com/osxfuse/osxfuse/issues/616), and we don't allow non-open source dependencies in Homebrew/core. If you need formulae that depend on macFUSE, try https://github.com/gromgit/homebrew-fuse.

**Reply by porg** [reply]
Thanks for the explanation! Thanks for the workaround!
@SMillerDev explained the key phrase "disabled". Nevertheless as someone tech-savvy but not ultra-deep in package-managers and FLOSS I could have never made sense of that message alone and deduce the cause of it.

Could you rephrase the message from:
> Disabled because it requires FUSE!

To something like:
> Disabled because it requires macFUSE.
macFUSE became closed source and hence got delisted from HomeBrew.
As an alternative you may use https://github.com/gromgit/homebrew-fuse.

**Reply by nandahkrishna** [reply]
A contributor created a pull request (Homebrew/homebrew-core#77289) to modify the message but it was closed as stale (and not merged before that because it wasn't updated based on suggestions). You're welcome to create a new PR to propose changes to the message. However, note that we do not "officially" link to third-party taps because we have no control over them.

For what it's worth, there are _many_ issues and discussions related to the disabling of FUSE-related formulae, including one I remember from a couple of days back (Homebrew/brew#11621).

**Reply by gromgit** [reply]
> A contributor created a pull request ([Homebrew/homebrew-core#77289](https://github.com/Homebrew/homebrew-core/pull/77289)) to modify the message but it was closed as stale

I've initiated https://github.com/Homebrew/homebrew-core/pull/80362 to follow up.

**Comment by SMillerDev** [comment]
> curlftpfs has been disabled

This is the key phrase here. It's not failing. It's disabled.

MacFUSE is no longer open source so homebrew does not support it anymore.

**Reply by porg** [reply]
Thanks for the explanation. Though hard to understand for someone occasionally using brew but not familar with it in depth. See proposal in the thread of @carlocab
